scripts: west: explicitly ask for --sectorerase for nRF9160_pca10090

ioannisg opened the pull request:
When flashing nRF9160_pca10090 board with nrfjprog explicitly
request for --sectorerase, instead of --sectoranduicrerase.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
scripts: west: explicitly ask for --sectorerase for nRF9160_pca10090

When flashing nRF9160_pca10090 board with nrfjprog, explicitly
request for --sectorerase, instead of --sectoranduicrerase (used
when flashing boards with nRF52 SOC).

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

carlescufi reviewed (changes requested):
Looks good, but please add a copyright line

ioannisg commented:
> 
> 
> Looks good, but please add a copyright line

done

mbolivar commented on scripts/west_commands/runners/nrfjprog.py:
```diff
@@ -104,10 +105,10 @@ def do_run(self, command, **kwargs):
-            else:
+            if self.family == 'NRF52':
                 commands.append(program_cmd + ['--sectoranduicrerase'])
+            else:
```
Hm... this should break tests cases and I'm not sure why it's not.

```
    '9YNN':
    (['nrfjprog', '--program', RC_KERNEL_HEX, '-f', 'NRF91', '--snr', TEST_DEF_SNR, '--sectoranduicrerase'],  # noqa: E501
     ['nrfjprog', '--reset', '-f', 'NRF91', '--snr', TEST_DEF_SNR]),
```

carlescufi replied:
This is now fixed, thanks for catching it!

mbolivar reviewed (changes requested):
Holding the merge while we're figuring out the testing situation

mbolivar commented:
@ioannisg this is the problem:

```
TEST_CASES = [(f, sr, snr, e)
             for f in ('NRF51', 'NRF52')
             for sr in (False, True)
             for snr in (TEST_OVR_SNR, None)
             for e in (False, True)]
```

Can you fix this and fix the tests for this PR?

carlescufi commented:
@ioannisg I will fix this, no need for you to set this up now

carlescufi committed:
west: runners: nrfjprog: Correct nRF91 unit test reference data

nRF91 only does `--sectorerase` (no UICR) and does not need to
enable pin reset (`--pinresetenable`).

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
west: runners: nrfjprog: Enable testing of nRF91

Enable the nRF91 suite of tests.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented:
@mbolivar 	all done, thanks!
